### ISSUE TITLE
unable to execute /usr/bin/katoolin

### ISSUE THREAD

**Issue by nathmo** [LionSec/katoolin#315]
unable to execute /usr/bin/katoolin

the shebang was set to python but python was updated and should explicitly specify python2
if you manually fix that by edditing the first line and running "sudo apt install python2" you might run into 

unable to execute /usr/bin/katoolin: No such file or directory
to fix that you should replace the line 53 by 
"apt-key adv --keyserver --keyserver keyserver.ubuntu.com --recv-keys ED444FF07D8D0BF6"

since this tool seems to be no longer maintained i forked it and made those two modification.

here is how to fix it without manually patching the script
'''
sudo su
'''
'''
apt install python2
'''
'''
git clone https://github.com/nathmo/katoolin.git && cp katoolin/katoolin.py /usr/bin/katoolin
'''
'''
chmod +x /usr/bin/katoolin
'''
'''
sudo katoolin 
'''

**Comment by Topplehawk** [comment]
yes it did, also for anyone reading this you should know that it is best to start over from the beginning but it will say that it already exists, but because it is a locked file since you typed in "sudo su" at the beginning, this means that it wont let you delete it. to deal with this open up terminal and type in "sudo rm -rf" then drag the file into the terminal and hit enter. this will delete the file for you.

**Comment by nathmo** [comment]
there. updated the instuction if you attempted to install the broken version first.
`
sudo su
`
`
rm -rf katoolin
`
`
rm /usr/bin/katoolin
`
`
apt install python2
`
`
git clone https://github.com/nathmo/katoolin.git && cp katoolin/katoolin.py /usr/bin/katoolin
`
`
chmod +x /usr/bin/katoolin
`
`
exit
`

`
sudo katoolin 
`

**Comment by Topplehawk** [comment]
> You do not have the permissions necessary to save the file. Please check that you typed the location correctly and try again

yes this is what i am refering to

**Comment by doc197** [comment]
worked for me ....stupid PYTHON!

**Comment by Rexmarek** [comment]
> there. updated the instuction if you attempted to install the broken version first. (it assume your shell / terminal is in the same path where you ran the command the first time. if not. after writing rm -rf but before pressing enter drag and drop the folder to the terminal so that it add the correct path to delete it. `sudo su` `rm -rf katoolin` `rm /usr/bin/katoolin` `apt install python2` `git clone https://github.com/nathmo/katoolin.git && cp katoolin/katoolin.py /usr/bin/katoolin` `chmod +x /usr/bin/katoolin` `exit`
> 
> `sudo katoolin `

Some people will have to do the following (after inputting the code above):
`cd /usr/bin`
`sudo nano katoolin`
Change `#!/usr/bin/python` to `#!/usr/bin/python2` then press control+x, press enter to write input buffer, press enter to write file name.
Run: `sudo ./katoolin`

Should work after that :)

**Comment by Kingki19** [comment]
Thank you very much

**Comment by Rvaldebon** [comment]
Good morning,

I have a problem for a moment to run option 1 "add kali linux repositories"

An error message tells me:

Warning: apt-key is deprecated. Manage keyring files in trusted.gpg.d instead (see apt-key(8)).
Executing: /tmp/apt-key-gpghome.CsBIKXJgfT/gpg.1.sh --keyserver pool.sks-keyservers.net --recv-keys ED444FF07D8D0BF6
gpg: failed to receive from keyserver: Server indicated a failure

If anyone has a solution, I'm interested !

**Comment by nathmo** [comment]
Hi, did you try my fork ? or manually modified the version version from @LionSec  ?
what OS and what version are you running ?
I currently have no time to look into it but if you can wait it might be fixed in a few weeks.

**Comment by tntxtra** [comment]
apt-key deprecated would love to try this out. I cloned from yours.

**Comment by oelja** [comment]
Thank you! This worked for me on a fresh install of Linux Mint after several hours of making assumptions about python!
